src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,216 @@
+    private static final String LOGIN_WITH_SPACES = "asdfg hjklfg sdcv";
+    private static final String MIN_PASSWORD_LENGTH = "01t2t3#";
+    private static final String MAX_PASSWORD_LENGTH = "qwertyuiop[]#$%^&asdfghj";
+    private static final String WITHOUT_SPECIAL_SYMBOLS = "password78909876";
+    private static final int MIN_AGE = 18;
+    private static final int MAX_AGE = 110;
+    private static final int NEGATIVE_AGE = -18;
+    private static final String MIN_LOGIN_LENGTH = "abcdef";
+    private static final String LESS_THAN_MIN_LOGIN_LENGTH = "abcde";
+    private static final int SMALLER_THEN_MIN_AGE = 17;
+    private static final String EMPTY_PASSWORD = "";
+
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setId(MIN_ID_LENGTH);
+        user.setLogin(LOGIN_OK);
+        user.setPassword(MIN_PASSWORD_LENGTH);
+        user.setAge(MIN_AGE);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void registerUser_NullId_notOk() {
+        user.setId(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void registerUser_InvalidId_LessThanMinLength_NotOk() {
+        user.setId(MIN_ID_LENGTH - 1);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void registerUser_InvalidId_GreaterThanMaxLength_NotOk() {
+        user.setId(MAX_ID_LENGTH + 1);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void registerUser_NullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void registerUser_InvalidLoginFormat_LessThanMinLength_notOk() {
+        user.setLogin(LESS_THAN_MIN_LOGIN_LENGTH);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void registerUser_InvalidLoginFormat_LoginWithSpaces_notOk() {
+        user.setLogin(LOGIN_WITH_SPACES);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void registerUser_EmptyPassword_notOk() {
+        user.setPassword(EMPTY_PASSWORD);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void registerUser_InvalidPasswordFormat_LessThanMinLength_notOk() {
+        user.setPassword(MIN_PASSWORD_LENGTH.substring(0, 3));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void registerUser_InvalidPasswordFormat_MoreThenMaxLength_notOk() {
+        user.setPassword(MAX_PASSWORD_LENGTH + 1);
```

Comment:
The test `registerUser_InvalidPasswordFormat_MoreThenMaxLength_notOk` is not correctly implemented. The expression `MAX_PASSWORD_LENGTH + 1` does not create a string longer than the maximum length. Please generate a string that exceeds `MAX_PASSWORD_LENGTH` to properly test this scenario.